homeassistant/components/launch_library/sensor.py

```diff
@@ -41,10 +52,56 @@
 )
 
 
-SENSOR_DESCRIPTION = SensorEntityDescription(
-    key="next_launch",
-    icon="mdi:rocket-launch",
-    name=DEFAULT_NAME,
+@dataclass
+class NextLaunchSensorEntityDescriptionMixin:
+    """Mixin for required keys."""
+
+    value_fn: Callable[[Launch], datetime | int | str | None]
+    attributes_fn: Callable[[Launch], dict[str, Any] | None]
+
+
+@dataclass
+class NextLaunchSensorEntityDescription(
+    SensorEntityDescription, NextLaunchSensorEntityDescriptionMixin
+):
+    """Describes a Next Launch sensor entity."""
+
+
+SENSOR_DESCRIPTIONS: tuple[NextLaunchSensorEntityDescription, ...] = (
+    NextLaunchSensorEntityDescription(
+        key=NEXT_LAUNCH,
+        icon="mdi:rocket-launch",
+        name=DEFAULT_NAME,
+        value_fn=lambda next_launch: next_launch.name,
+        attributes_fn=lambda next_launch: {
+            ATTR_LAUNCH_PROVIDER: next_launch.launch_service_provider.name,
+            ATTR_LAUNCH_PAD: next_launch.pad.name,
+            ATTR_LAUNCH_FACILITY: next_launch.pad.location.name,
+            ATTR_LAUNCH_PAD_COUNTRY_CODE: next_launch.pad.location.country_code,
+        },
+    ),
+    NextLaunchSensorEntityDescription(
+        key=LAUNCH_TIME,
+        icon="mdi:clock-outline",
+        name="Launch time",
+        device_class=SensorDeviceClass.TIMESTAMP,
+        value_fn=lambda next_launch: parse_datetime(next_launch.net),
+        attributes_fn=lambda next_launch: {
+            ATTR_WINDOW_START: next_launch.window_start,
+            ATTR_WINDOW_END: next_launch.window_end,
+            ATTR_STREAM_LIVE: next_launch.webcast_live,
+        },
+    ),
+    NextLaunchSensorEntityDescription(
+        key=LAUNCH_PROBABILITY,
+        icon="mdi:dice-multiple",
+        name="Launch Probability",
+        native_unit_of_measurement=PERCENTAGE,
+        value_fn=lambda next_launch: next_launch.probability
+        if next_launch.probability != -1
+        else STATE_UNKNOWN,
```

Comment:
We use `None` to represent unknown state.

Reply:
We prefer to define a regular function at the module level instead of the lambda if the lambda spans more than one line.

Reply:
https://github.com/home-assistant/core/pull/64424